doc.go

```diff
@@ -0,0 +1,32 @@
+// The [surrealdb] package implements [SurrealDB RPC Protocol] in the Go way.
```

Comment:
This will appear above the function index at https://pkg.go.dev/github.com/surrealdb/surrealdb.go#section-documentation so that the reader can go through (1) README and (2) package doc, to (3) each essential function doc in this order.